src/types/proceedings.d.ts

```diff
@@ -0,0 +1,31 @@
+import type { ResponseType, RequestType, TransportMedium } from '../types/request';
+import { PROCEEDING_STATUS } from '../Utility/requests';
+
+type MessageId = string;
+
+type Proceeding = {
+    reference: string;
+    messages: Record<MessageId, Message>;
+    status: ProceedingStatus;
+};
+
+type ProceedingStatus = typeof PROCEEDING_STATUS[string];
+
+type Message = {
+    id?: MessageId;
+    reference: string;
+    date: Date;
+    type: RequestType | ResponseType | 'response';
+    slug: string;
+    recipient: string;
+    email: string;
+    transport_medium: TransportMedium;
+    subject?: string;
+    content?: string;
+    attachments?: Attachement;
```

Comment:
Yes, i would have saved it as an attachment.

Reply:
But wouldn't we then have to distinguish between "real" and these attachments? If we want to display an attachments list, wouldn't it be weird if that also contains these fake attachments?

Reply:
So you propose a `message_file` key? But .eml files contain the attachments as well, don't they? I don't know. Maybe `content` should either be a string or a file?

Reply:
~~I don't know what the best way to go about this is.~~ Actually… when do we ever have separately stored attachments? As you said, they are already contained in the .eml, and for PDFs, I guess everything would also be in the same file.

Reply:
Right. Maybe there would be a case where people would want that, but I think those users can be trusted to append PDFs. But what does that mean? Just one `file` key?

Reply:
Guess so. Not sure whether we'd want to additionally store filename and subject.

But we actually don't have to decide that now either, since none of this matters for the website. We can always add additional properties when we know what we actually need.